docs/overview/image.rst

```diff
@@ -296,13 +297,56 @@ open/close the PNG files. It should look like this:
     * Decode an area of image
     * @param decoder      pointer to the decoder where this function belongs
     * @param dsc          image decoder descriptor
-    * @param full_area    full image area information
-    * @param decoded_area area information to decode (x1, y1, x2, y2)
-    * @return             LV_RESULT_OK: no error; LV_RESULT_INVALID: can't decode image area
+    * @param full_area    input parameter. the full area to decode after enough subsequent calls
+    * @param decoded_area input+output parameter. set the values to `LV_COORD_MIN` for the first call and to reset decoding.
+    *                     the decoded area is stored here after each call.
+    * @return             LV_RESULT_OK: ok; LV_RESULT_INVALID: failed or there is nothing left to decode
     */
    static lv_result_t decoder_get_area(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * dsc,
                                     const lv_area_t * full_area, lv_area_t * decoded_area)
    {
+     /**
+     * If `dsc->decoded` is always set in `decoder_open` then `decoder_get_area` does not need to be implemented.
+     * If `dsc->decoded` is only sometimes set or never set in `decoder_open` then `decoder_get_area` is used to
+     * incrementally decode the image by calling it repeatedly until it returns `LV_RESULT_INVALID`.
```

Comment:
Please add something like this:
In the example below we will decode the image line by line, but the decoded area can have any shape and size depending on the requirements and the capabilities of the image decoder.